### PR TITLE
Making bytearray a commandline switch

### PR DESCRIPTION
**Pull request by jerinphilip**
- [x] Remove `bergamot-app-bytearray` (This is moot since TranslationModel is gone and service-cli-bytearray achieves the intended outcome).
- [x] Collapse `service-cli{,-bytearray}`. This is a command line switch now.
- [x] Update brt scripts accordingly.

There's a precision issue from #119, which will be taken up separately. For now, mac tests are restricted in BRT to have only one test.

### CODE REVIEW

File: src/translator/parser.h [hunk 1]
```diff
@@ -23,9 +23,13 @@ inline marian::ConfigParser createConfigParser() {
...
       "Maximum input tokens to be processed in a single sentence.", 128);
 
+  cp.addOption<bool>(
+      "--bytearray", "Bergamot Options",
+      "Enable loads through byte-arrays to enable testing of byte-array codepath (false by default)", false);
+
   cp.addOption<bool>(
       "--check-bytearray", "Bergamot Options",
-      "Flag holds whether to check the content of the bytearray (true by default)", true);
+      "Flag holds whether to check the content of the bytearray (false by default)", false);
```
Comment (jerinphilip) [inline src/translator/parser.h:32]: @qianqianzhu I changed the default behaviour here. Setting `--bytearray` to false and `--check-bytearray` default to true leads to validation issues. So to ensure that the cli always also validates the blobs, I manually change the parsed option to true in the other place (https://github.com/browsermt/bergamot-translator/pull/127/files#r626677517).

Comment (XapaJIaMnu) [reply src/translator/parser.h:32]: Can we get a better solution. I don't like the hardcoded  `  options->set<bool>("check-bytearray", true);` even if it's just in the test binary.

Comment (qianqianzhu) [reply src/translator/parser.h:32]: I see your point. What I meant https://github.com/browsermt/bergamot-translator/pull/127#discussion_r626677517 here is that we do not need to manually set `--check-bytearray` to true.

File: app/service-cli.cpp [hunk 2]
```diff
@@ -14,7 +15,21 @@
...
+  // Prepare memories for model and shortlist
+  marian::bergamot::AlignedMemory modelBytes, shortlistBytes;
+
+  if (options->get<bool>("bytearray")) {
+    // Load legit values into bytearrays.
+    modelBytes = marian::bergamot::getModelMemoryFromConfig(options);
+    shortlistBytes = marian::bergamot::getShortlistMemoryFromConfig(options);
+
+    // Service command line always has check-bytearray on.
+    options->set<bool>("check-bytearray", true);
```
[the remark this answers is not in the document]
Comment (jerinphilip) [reply app/service-cli.cpp]: I have changed this to use `--check-bytearray`. Default being always check and default being loaded from bytearray. The scripts have explicit arguments to match old behaviour. Is this ok?

File: src/translator/parser.h (hunk 1 above)
Comment (jerinphilip) [reply src/translator/parser.h]: @XapaJIaMnu Take a look now?

**Review by qianqianzhu — APPROVED** [review]

**Review by XapaJIaMnu — APPROVED** [review]

**Pull request merged**